doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -154,6 +138,52 @@ each replica, including the times at which it was created and dropped
 | `dropped_at`          | [`timestamp with time zone`] | The time at which the replica was dropped, or `NULL` if it still exists.                                                                  |
 | `credits_per_hour`    | [`numeric`]                  | The number of compute credits consumed per hour. Corresponds to [`mz_cluster_replica_sizes.credits_per_hour`](#mz_cluster_replica_sizes). |
 
+### `mz_frontiers`
+
+The `mz_frontiers` table describes the frontiers of each source, sink, table,
+materialized view, index, and subscription in the system, as observed from the
+coordinator.
+
+For objects that are installed on replicas (e.g., materialized views and
+indexes), the `replica_id` field is always non-`NULL`. If an object is installed
+on multiple replicas, it has multiple entries describing the frontier on each
+individual replica. For objects that are not installed on replicas (e.g.,
+tables), the `replica_id` field is `NULL`.
+
+If you don't care about per-replica frontiers, using the
+[`mz_global_frontiers`](#mz_global_frontiers) view instead is more convenient.
+
+[`mz_compute_frontiers`](#mz_compute_frontiers) is similar to this table, but
+`mz_compute_frontiers` reports the frontiers known to the active compute
+replica, while this table reports the frontiers the coordinator is aware of.
+Note also that `mz_compute_frontiers` is restricted to compute objects (indexes,
+materialized views, and subscriptions) while this table contains storage objects
+(sources, sinks, and tables) as well.
+
+At this time, we do not make any guarantees about the freshness of these numbers.
+
+<!-- RELATION_SPEC mz_internal.mz_frontiers -->
+| Field         | Type             | Meaning                                                                                       |
+| ------------- | ------------     | --------                                                                                      |
+| `object_id`   | [`text`]         | The ID of the source, sink, table, index, materialized view, or subscription.                 |
+| `replica_id`  | [`text`]         | The ID of a cluster replica, or `NULL` if the object is not installed on a replica.           |
+| `time`        | [`mz_timestamp`] | The next timestamp at which the output may change, or `NULL` if all output has been produced. |
```

Comment:
This uses the same naming as `mz_compute_frontiers`: https://materialize.com/docs/sql/system-catalog/mz_internal/#mz_compute_frontiers

Generally, a frontier can consist of multiple times. I think `mz_compute_frontier` was initially written with the intention that it would contain one row per time in the frontier. You would then reconstruct a frontier by `SELECT time FROM mz_compute_frontiers WHERE id = ...`. But every frontier that compute and storage export today consists of totally ordered times, so there is either a single or no time in the frontier, which means you either have one or zero rows.

It seems fine to rename to `frontier`, as you suggested. But if we do that, we should also rename the other relations that expose frontiers for consistency. I think this warrants a separate PR.

Reply:
I don't have a strong preference on `frontier` vs `time`, just throwing the idea out in case someone else does. Your explanation of one row per time makes sense to me though and I think justifies calling it `time`.